search/indexing_api.py

```diff
@@ -125,6 +129,68 @@ def _delete_item(document_id, doc_type, index):
         pass
 
 
+def serialize_public_enrolled_user(serialized_enrolled_user):
+    """
+    Creates a public serialized version of an enrolled user.
+
+    Args:
+        serialized_enrolled_user(dict):
+            Serialized enrolled user generated by serialize_program_enrolled_user
+
+    Returns:
+        dict:
+            serialized subset of the original document sanitized for public consumption
+    """
+    # filter out grades, courses passed, etc
+    program = dict_with_keys(
+        serialized_enrolled_user['program'],
+        ['id', 'enrollments', 'is_learner', 'total_courses', ]
+    )
+    program['enrollments'] = [
+        dict_with_keys(enrollment, ['course_title', 'semester', ])
+        for enrollment in program['enrollments']
+    ]
+    # filter out private profile information
+    profile = dict_with_keys(
+        serialized_enrolled_user['profile'],
+        [
+            'first_name', 'last_name', 'preferred_name',
+            'romanized_first_name', 'romanized_last_name',
+            'image', 'image_small', 'image_medium',
+            'username', 'filled_out', 'account_privacy',
+            'country', 'state_or_territory', 'city',
+            'birth_country', 'work_history',
+        ]
+    )
+    return {
+        'id': serialized_enrolled_user['id'],
+        '_id': serialized_enrolled_user['_id'],
+        '_type': PUBLIC_USER_DOC_TYPE,  # override the default of USER_DOC_TYPE
+        'user_id': serialized_enrolled_user['user_id'],
+        'profile': profile,
+        'program': program,
+    }
+
+
+def _get_program_enrolled_users_documents(program_enrollments):
+    """
+    Generator for documents to be indexed given a set of program enrollments
+    Args:
+        program_enrollments(iterable of ProgramEnrollment):
+            iterable of enrollments to generate documents for
+    Yields:
+        for each enrollment:
+            a private (staff-only search) document
+            a public (learner-learner search) document
+    """
+    for program_enrollment in program_enrollments:
+        document = serialize_program_enrolled_user(program_enrollment)
+        if document is None:
```

Comment:
:sob: testament to the longevity of this PR https://github.com/mitodl/micromasters/pull/2908#issuecomment-294157138

https://github.com/mitodl/micromasters/pull/2908/files#diff-f53b42083d3ee738beef375b588e8a02L201